services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,54 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
+    if (event == NULL) {
+        return ERR_CODE_INVALID_QUEUE_MSG;}
+    if (xQueueSend(thermalMgrQueueHandle, (void *)event, (TickType_t) 0) == pdTRUE){
+        return ERR_CODE_SUCCESS;
+    }
+  return ERR_CODE_UNKNOWN; // prev: ERR_CODE_QUEUE_FULL;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt;
+  interrupt.type = THERMAL_MGR_EVENT_INTERRUPTS;
+    thermalMgrSendEvent(&interrupt);
+
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+    thermal_mgr_event_t thermalReceivedData;
+    float currentTemperature;
```

Comment:
In general, try to limit the scope of a variable as much as possible. For example, neither of these variables are used outside the `while` loop, so they can be declared inside the loop.